File: Video Player/NSString+Base64.m
//
//  NSString+Base64.m
//  Video Player
//
//  Created by 朱 文杰 on 15/8/1.
//  Copyright (c) 2015年 Home. All rights reserved.
//

#import "NSString+Base64.h"

@implementation NSString (Base64)
- (NSString *)base64String {
    return [[self dataUsingEncoding:NSUTF8StringEncoding] base64EncodedStringWithOptions:0];
}
- (NSString *)decodedBase64String {
    return [[NSString alloc] initWithData:[[NSData alloc] initWithBase64EncodedString:self options:0] encoding:NSUTF8StringEncoding];
}
@end

File: Video Player/NSString+Base64.swift
//
//  NSString+Base64.swift
//  Video Player
//
//  Created by 朱文杰 on 15/11/2.
//  Copyright © 2015年 Home. All rights reserved.
//

import Foundation

@available(iOS 7.0, OSX 10.9, *)
public extension NSString {
    func base64String() -> NSString? {
        guard let data = self.dataUsingEncoding(NSUTF8StringEncoding) else { return nil }
        return data.base64EncodedStringWithOptions([])
    }

    func decodedBase64String() -> NSString? {
        guard let data = NSData(base64EncodedString: (self as String), options: []) else { return nil }
        return NSString(data: data, encoding: NSUTF8StringEncoding)
    }
}
